pkg/controllers/node/termination/terminator/terminator.go

```diff
@@ -46,18 +46,12 @@ func NewTerminator(clk clock.Clock, kubeClient client.Client, eq *Queue) *Termin
 }
 
 // Taint adds the karpenter.sh/disruption taint to a node with a NodeClaim
-// If the node is linked to a machine, it will use the node.kubernetes.io/unschedulable taint.
 func (t *Terminator) Taint(ctx context.Context, node *v1.Node) error {
 	stored := node.DeepCopy()
-
-	if _, isMachine := node.Labels[v1alpha5.ProvisionerNameLabelKey]; isMachine {
-		node.Spec.Unschedulable = true
-	} else {
-		node.Spec.Taints = lo.Reject(node.Spec.Taints, func(t v1.Taint, _ int) bool {
-			return v1beta1.IsDisruptingTaint(t)
-		})
-		node.Spec.Taints = append(node.Spec.Taints, v1beta1.DisruptionNoScheduleTaint)
-	}
+	node.Spec.Taints = lo.Reject(node.Spec.Taints, func(t v1.Taint, _ int) bool {
```

Comment:
Maybe a comment since this logic might be a little confusing to someone just looking at it. One concern looking at this code: Does it have the potential to re-order taints if someone appended their own taint to the end after we taint it? Should we just handle the add with a single range loop so we don't have the potential to affect the order?